mlflow/models/evaluation/base.py

```diff
@@ -778,6 +798,45 @@ def evaluate(
                              If multiple evaluators are specified, each configuration should be
                              supplied as a nested dictionary whose key is the evaluator name.
 
+    :param custom_metrics: (Optional) A list of custom metric functions. A custom metric
+                              function is required to take in two parameters:
+                              - Union[pandas.Dataframe, pyspark.sql.DataFrame]: The first being a
+                                Pandas or Spark DataFrame containing ``prediction`` and ``target``
+                                column. The ``prediction`` column contains the predictions made by
+                                the model. The ``target`` column contains the corresponding labels
+                                to the predictions made on that row.
+                              - Dict: The second is a dictionary containing the metrics calculated
+                                by the default evaluator. The keys are the names of the metrics
+                                and the values are the scalar values of the metrics. Refer to the
+                                DefaultEvaluator behavior section for what metrics will be returned
+                                based on the type of model (i.e. classifier or regressor).
+
+                              A custom metric function can return in two different formats:
+                              - Dict[AnyStr, Union[int, float, np.number]: a singular dictionary of
+                                custom metrics, where the keys are the names of the metrics, and the
+                                values are the scalar values of the metrics.
+
+                              .. code-block:: python
+                                  :caption: Custom Metric Function Boilerplate:
+                                  def custom_metrics_boilerplate(eval_df, builtin_metrics):
+                                      ...
+                                      metrics: Dict[AnyStr, Union[int, float, np.number]] = ???
+                                      ...
+                                      return metrics
+
+                              .. code-block:: python
+                                  :caption: Example usage of custom metrics:
+                                  def squared_diff_plus_one(eval_df, builtin_metrics):
+                                      return {
+                                          "sdiff_plus_one": (np.sum(np.abs(eval_df["prediction"]
+                                                            - eval_df["target"]+1)**2))
+                                      }
+
+                                  with mlflow.start_run():
+                                      mlflow.evaluate(
+                                          model, X, targets, custom_metrics=[
+                                          squared_diff_plus_one,...])
```

Comment:
Nit: Can we clean up the formatting here? Perhaps running this through `black` would be helpful.